libs/language-server/src/lib/ast/wrappers/value-type/valuetype.ts

```diff
@@ -78,10 +93,18 @@ export abstract class AbstractValuetype implements Valuetype {
     return this.supertype;
   }
 
+  equals(target: Valuetype): boolean {
+    return this.isConvertibleTo(target) && target.isConvertibleTo(this); // TODO: solve with registry and === comparison instead
```

Comment:
We should replace this by a registry mechanism that enforces singletons and compare with `===`